src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -157,20 +146,36 @@ class EasyFedora2vaultApp(configuration: Configuration) extends DebugEnhancedLog
   }
 
   private def addXmlPayload(bag: DansV0Bag, path: String)(content: Node): Try[Any] = {
-    bag.addTagFile(content.serialize.inputStream, Paths.get(path))
+    bag.addPayloadFile(content.serialize.inputStream, Paths.get(path))
   }
 
   private def addPayloadFileTo(bag: DansV0Bag)(fedoraFileId: String): Try[FileItem] = {
-    fedoraProvider.loadFoXml(fedoraFileId)
-      .flatMap { foXml =>
-        val metadata = foXml \\ "file-item-md"
-        val path = Paths.get((metadata \\ "path").text)
-        logger.info(s"Adding $fedoraFileId to $path")
-        fedoraProvider
-          .disseminateDatastream(fedoraFileId, "EASY_FILE")
-          .map(bag.addPayloadFile(_, path))
-          .tried.flatten
-          .flatMap(_ => FileItem(fedoraFileId, foXml))
-      }
+    for {
+      foXml <- fedoraProvider.loadFoXml(fedoraFileId)
+      path = Paths.get((foXml \\ "file-item-md" \\ "path").text)
+      _ = logger.info(s"Adding $fedoraFileId to $path")
+      _ <- fedoraProvider
+        .disseminateDatastream(fedoraFileId, "EASY_FILE")
+        .map(bag.addPayloadFile(_, path))
+        .tried.flatten
+      digest = foXml \ "datastream" \ "datastreamVersion" \ "contentDigest"
+      _ <- validate(digest, bag.baseDir / s"data/$path", bag, fedoraFileId)
+      fileItem <- FileItem(fedoraFileId, foXml)
+    } yield fileItem
+  }
+
+  private def validate(digest: NodeSeq, file: File, bag: DansV0Bag, fedoraFileId: String) = Try {
+    // note that the contentDigest is found in different streams
+    // such as streamId: EASY_FILE and EASY_FILE_METADATA
```

Comment:
On deasy the situation is even worse, some files have no digest at all, others multiple as shown below.
@lindareijnhoudt what is the situation on production?
I guess I should only use the digest on the `datastream` with id `EASY_FILE`, if provided.

![image](https://user-images.githubusercontent.com/10553630/82294728-e1361a80-99ae-11ea-8adb-6f7349c2d4ae.png)

Reply:
the objects i saw in EASY11 are indeed like this, so the datastream with id EASY_FILE and the contentDigest therein